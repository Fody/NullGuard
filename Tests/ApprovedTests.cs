#if (NET472)
using VerifyXunit;
using Fody;
using Xunit;
using System.Linq;
using System.Threading.Tasks;

[UsesVerify]
public class ApprovedTests
{
    [Fact]
    public Task ClassWithBadAttributes()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithBadAttributes"));
    }

    [Fact]
    public Task ClassWithPrivateMethod()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithPrivateMethod"));
    }

    [Fact]
    public Task ClassWithPrivateMethodNoAssert()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithPrivateMethod"));
    }

    [Fact]
    public Task GenericClass()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "GenericClass`1"));
    }

    [Fact]
    public Task GenericClassWithValueTypeConstraint()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "GenericClassWithValueTypeConstraints`1"));
    }

    [Fact]
    public Task GenericClassWithReferenceTypeConstraints()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "GenericClassWithReferenceTypeConstraints`1"));
    }

    [Fact]
    public Task Indexers()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "Indexers"));
    }

    [Fact]
    public Task InterfaceBadAttributes()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "InterfaceBadAttributes"));
    }

    [Fact]
    public Task SimpleClass()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleClass"));
    }

    [Fact]
    public Task SimpleClassNoAssert()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleClass"));
    }

    [Fact]
    public Task SkipIXamlMetadataProvider()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "XamlMetadataProvider"));
    }

#if (DEBUG)
    [Fact]
    public Task SpecialClass_debug()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SpecialClass"));
    }
#else
    [Fact]
    public Task SpecialClass_release()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SpecialClass"));
    }
#endif

    [Fact]
    public Task PublicNestedInsideNonPublic()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "NonPublicWithNested"));
    }

    [Fact]
    public Task UnsafeClass()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "UnsafeClass"));
    }

    [Fact]
    public Task ClassWithImplicitInterface()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithImplicitInterface"));
    }

    [Fact]
    public Task ClassWithExplicitInterface()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithExplicitInterface"));
    }

    [Fact]
    public Task InfosList()
    {
        return Verifier.Verify(AssemblyWeaver.TestResult.Messages.Select(x=>x.Text));
    }

    [Fact]
    public Task WarnsList()
    {
        return Verifier.Verify(AssemblyWeaver.TestResult.Warnings.Select(x=>x.Text));
    }

    [Fact]
    public Task ErrorsList()
    {
        return Verifier.Verify(AssemblyWeaver.TestResult.Errors.Select(x=>x.Text));
    }
}

#endif