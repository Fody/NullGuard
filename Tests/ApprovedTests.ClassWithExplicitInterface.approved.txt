.class public auto ansi beforefieldinit ClassWithExplicitInterface
       extends [mscorlib]System.Object
       implements class [mscorlib]System.IComparable`1<string>
{
  .class interface abstract auto ansi nested private IPrivate
  {
    .method public hidebysig newslot abstract virtual 
            instance int32  CompareTo(string other) cil managed
    {
    } // end of method IPrivate::CompareTo
  } 
  .class auto ansi nested private beforefieldinit ClassWithExplicitPrivateInterface
         extends [mscorlib]System.Object
         implements ClassWithExplicitInterface/IPrivate
  {
    .method private hidebysig newslot virtual final 
            instance int32  ClassWithExplicitInterface.IPrivate.CompareTo(string other1) cil managed
    {
      .override ClassWithExplicitInterface/IPrivate::CompareTo
      // Code size       2 (0x2)
      .maxstack  8
      .language '{3F5162F8-07C6-11D3-9053-00C04FA302A1}', '{994B45C4-E6E9-11D2-903F-00C04FA302A1}', '{5A869D0B-6611-11D3-BD2A-0000F80849BD}'
//000032:             return 0;
      IL_0000:  ldc.i4.0
      IL_0001:  ret
    } // end of method ClassWithExplicitPrivateInterface::ClassWithExplicitInterface.IPrivate.CompareTo
    .method public hidebysig specialname rtspecialname 
            instance void  .ctor() cil managed
    {
      // Code size       7 (0x7)
      .maxstack  8
      IL_0000:  ldarg.0
      IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
      IL_0006:  ret
    } // end of method ClassWithExplicitPrivateInterface::.ctor
  } 
  .class auto ansi nested private beforefieldinit ClassWithExplicitPublicInterface
         extends [mscorlib]System.Object
         implements class [mscorlib]System.IComparable`1<string>
  {
    .method private hidebysig newslot virtual final 
            instance int32  'System.IComparable<System.String>.CompareTo'(string other1) cil managed
    {
      .override  method instance int32 class [mscorlib]System.IComparable`1<string>::CompareTo(!0)
      // Code size       21 (0x15)
      .maxstack  2
//000033:         }
//000034:     }
//000035: 
//000036:     private class ClassWithExplicitPublicInterface : IComparable<string>
//000037:     {
//000038:         int IComparable<string>.CompareTo(string other1)
//000039:         {
//000040:             return 1;
//000041:         }
//000042:     }
//000043: }
      IL_0000:  ldarg.1
      IL_0001:  brtrue.s   IL_0013
      IL_0003:  ldstr      "other1"
      IL_0008:  ldstr      "[NullGuard] other1 is null."
      IL_000d:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
                                                                                       string)
      IL_0012:  throw
//000040:             return 1;
      IL_0013:  ldc.i4.1
      IL_0014:  ret
    } // end of method ClassWithExplicitPublicInterface::'System.IComparable<System.String>.CompareTo'
    .method public hidebysig specialname rtspecialname 
            instance void  .ctor() cil managed
    {
      // Code size       7 (0x7)
      .maxstack  8
      IL_0000:  ldarg.0
      IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
      IL_0006:  ret
    } // end of method ClassWithExplicitPublicInterface::.ctor
  } 
  .method private hidebysig newslot virtual final 
          instance int32  'System.IComparable<System.String>.CompareTo'(string other) cil managed
  {
    .override  method instance int32 class [mscorlib]System.IComparable`1<string>::CompareTo(!0)
    // Code size       21 (0x15)
    .maxstack  2
//000041:         }
//000042:     }
//000043: }
    IL_0000:  ldarg.1
    IL_0001:  brtrue.s   IL_0013
    IL_0003:  ldstr      "other"
    IL_0008:  ldstr      "[NullGuard] other is null."
    IL_000d:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
                                                                                     string)
    IL_0012:  throw
//000009:         return 0;
    IL_0013:  ldc.i4.0
    IL_0014:  ret
  } 
  .method public hidebysig instance int32 
          CallInteralClassWithPrivateInterface(string other) cil managed
  {
    // Code size       12 (0xc)
    .maxstack  2
//000010:     }
//000011: 
//000012:     public int CallInteralClassWithPrivateInterface([AllowNull] string other)
//000013:     {
//000014:         return ((IPrivate) new ClassWithExplicitPrivateInterface()).CompareTo(other);
    IL_0000:  newobj     instance void ClassWithExplicitInterface/ClassWithExplicitPrivateInterface::.ctor()
    IL_0005:  ldarg.1
    IL_0006:  callvirt   instance int32 ClassWithExplicitInterface/IPrivate::CompareTo(string)
    IL_000b:  ret
  } 
  .method public hidebysig instance int32 
          CallInteralClassWithPublicInterface(string other) cil managed
  {
    // Code size       12 (0xc)
    .maxstack  2
//000015:     }
//000016: 
//000017:     public int CallInteralClassWithPublicInterface([AllowNull] string other)
//000018:     {
//000019:         return ((IComparable<string>) new ClassWithExplicitPublicInterface()).CompareTo(other);
    IL_0000:  newobj     instance void ClassWithExplicitInterface/ClassWithExplicitPublicInterface::.ctor()
    IL_0005:  ldarg.1
    IL_0006:  callvirt   instance int32 class [mscorlib]System.IComparable`1<string>::CompareTo(!0)
    IL_000b:  ret
  } 
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       7 (0x7)
    .maxstack  1
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
    IL_0006:  ret
  } 
}
